app/models/manageiq/providers/physical_infra_manager.rb

```diff
@@ -50,5 +50,51 @@ def console_supported?
     def console_url
       raise MiqException::Error, _("Console not supported")
     end
+
+    supports :change_password
+
+    # Changes the password of userId on provider client and database.
+    #
+    # @param [current_password] password currently used for connected userId in provider client
+    # @param [new_password]     password that will replace the current one
+    #
+    # @return [Boolean] true if the password was changed successfully
+    def change_password(current_password, new_password, auth_type = :default)
+      change_password_on_provider(current_password, new_password)
+      update_authentication(auth_type => {:userid => authentication_userid, :password => new_password})
+
+      true
+    end
+
+    def queue_change_password(userid, current_password, new_password, auth_type = :default)
+      task_opts = {
+        :action => "Changing the password for Physical Provider named '#{name}'",
+        :userid => userid
+      }
+
+      queue_opts = {
+        :class_name  => self.class.name,
+        :instance_id => id,
+        :method_name => 'change_password',
+        :priority    => MiqQueue::HIGH_PRIORITY,
```

Comment:
Just leave the priority out, the default is normal which is good enough for this.  We want to leave high priority open for tasks which must be run before others

Reply:
Right